components/courses/app/services/courses/student/index.rb

```diff
@@ -0,0 +1,23 @@
+module Courses
+  class Student < ApplicationRecord
+    class Index < ApplicationService
+      def initialize(season)
+        @season = season
+      end
+
+      def call
+        send "#{@season.status}_season_students".to_sym, @season
+      end
+
+      private
+
+      attr_reader :season
+
+      Courses::Season.statuses.keys.each do |status|
+        define_method :"#{status}_season_students" do |season|
+          "::Courses::Student::#{status.capitalize}Season".constantize.call(season)
```

Comment:
Cool. But redundant. We should avoid ths magic metaprogramming. 

Create only one simple finder `Courses::Seasons::StudentsPerStatus` and use simple `case` statements.
Easy life.